### PR TITLE
instructions for core-js publishing

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
- README improvements

### CODE REVIEW

**Comment by michaelneale** [comment]
:bee: that failure is the intermittent fav tests...

File: blueocean-core-js/README.md [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+- You should see all package.json and npm-shrinkwrap.json updated to reflect the new version number.
+- Push changes to your branch and ensure that the blueocean-plugin and ATH builds both pass.
+
+## Publishing a Production Version
+
+Once the changes to blueocean-core-js are approved via PR, perform the following steps:
+- Ensure your branch is 100% up to date with master. **This is critical**, otherwise recent changes may be lost and break the app.
+   - If changes to core-js were made in the interim, ideally you should merge and publish a new pre-release version to ensure builds still pass.
+- Tick version numbers in package.json and npm-shrinkwrap.json to a production version, e.g. "0.0.90"
+- `npm run bundle`
```
Comment (i386) [inline blueocean-core-js/README.md]: `npn run bundle` isn't a thing.

```
shallowblue:blueocean-core-js jdumay$ npm run bundle
npm ERR! Darwin 16.5.0
npm ERR! argv "/usr/local/Cellar/node/7.8.0/bin/node" "/usr/local/bin/npm" "run" "bundle"
npm ERR! node v7.8.0
npm ERR! npm  v4.2.0

npm ERR! missing script: bundle
npm ERR! 
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/jdumay/.npm/_logs/2017-04-19T22_25_15_013Z-debug.log
```

However `npm publish` will run tests, bundle and publish for you

**Comment by i386** [comment]
🐝  left a comment about the bundle command being missing from core otherwise LGTM

File: blueocean-core-js/README.md (hunk 1 above)
Comment (michaelneale) [inline blueocean-core-js/README.md]: odd, there is bundle:watch... but why no bundle? should it just be an alias to run gulp? Probably should correct the package.json. 

Alternatively, given the package.json is source of truth, perhaps we shouldn't repeat what is clearly in there?? (although it does look nicer) - so a bit of an :ant:

File: blueocean-core-js/README.md [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+
+## Developing with Prerelease Versions
+
+To test changes in blueocean-core-js, perform the following steps:
+- Tick the version number in package.json and npm-shrinkwrap.json and ensure a "prelease" version specified, e.g. 0.0.90-SNAPSHOT-1
+- `npm run gulp`
+- `npm publish --tag beta` (the "beta" tag is important)
+   - If you receive an error about the package already existing, tick the version up again, e.g. 0.0.90-SNAPSHOT-2 and repeat above steps.
+- Then perform the following in blueocean-web, blueocean-dashboard, and blueocean-personalization
+   - `npm install @jenkins-cd/blueocean-core-js@0.0.90-SNAPSHOT-1 -S -E`
```
Comment (scherler) [inline blueocean-core-js/README.md]: Why not include a reference to `bin/cleanInstall` which does that in parallel and can further build those with mvn

Comment (scherler) [inline blueocean-core-js/README.md]: https://github.com/jenkinsci/blueocean-plugin/blob/master/bin/cleanInstall.js

Comment (cliffmeyers) [inline blueocean-core-js/README.md]: Added in 11c8784. Could you verify my instructions are correct? I've been doing it by hand out of habit I guess :/

Comment (cliffmeyers) [inline blueocean-core-js/README.md]: @scherler look ok to you?

**Pull request merged**